tick/optim/solver/svrg.py

```diff
@@ -98,12 +103,38 @@ def variance_reduction(self):
 
     @variance_reduction.setter
     def variance_reduction(self, val: str):
-
         if val not in variance_reduction_methods_mapper:
             raise ValueError(
-                'variance_reduction should be one of "{}", got "{}".'.format(
-                    ', '.join(variance_reduction_methods_mapper.keys()),
+                'variance_reduction should be one of "{}", got "{}"'.format(
+                    ', '.join(sorted(variance_reduction_methods_mapper.keys())),
                     val))
-
+        if self.model is not None:
+            if val == 'avg' and self.model._model.is_sparse():
+                warn("'avg' variance reduction cannot be used "
+                     "with delayed updates for sparse data", UserWarning)
         self._solver.set_variance_reduction(
             variance_reduction_methods_mapper[val])
+
+    def set_model(self, model: Model):
+        """Set model in the solver
+
+        Parameters
+        ----------
+        model : `Model`
+            Sets the model in the solver. The model gives the first
+            order information about the model (loss, gradient, among
+            other things)
+
+        Returns
+        -------
+        output : `Solver`
+            The `Solver` with given model
+        """
+        # We need to check that the setted model is not sparse, while the
+        # variance reduction method is 'avg
+        if self.variance_reduction == 'avg' and model._model.is_sparse():
+            warn("'avg' variance reduction cannot be used with delayed updates "
```

Comment:
again, no need to mention delayed updates

Reply:
Done